packages/client/lib/sync/fullsync.ts

```diff
@@ -205,10 +213,62 @@ export class FullSynchronizer extends Synchronizer {
    * @param blockData `NEW_BLOCK` received from peer
    */
   async handleNewBlock(block: Block) {
-    const chainTip = (await this.chain.getLatestHeader()).hash()
+    const chainTip = await this.chain.getLatestBlock()
+
     // If block parent is current chain tip, insert block into chain
-    if (chainTip.toString('hex') === block.header.parentHash.toString('hex')) {
+    if (chainTip.header.hash().toString('hex') === block.header.parentHash.toString('hex')) {
```

Comment:
just FYI for the future, we replaced a lot of `buffer1.toString('hex') === buffer2.toString('hex')` in the monorepo previously since `toString` is a rather slow operation and comparison, the preferred faster way is to use `buffer1.equals(buffer2)`
```suggestion
    if (chainTip.header.hash().equals(block.header.parentHash)) {
```